bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -298,9 +303,11 @@ public void thingRemoved(DiscoveryService source, ThingUID thingUID) {
     public Collection<ThingUID> removeOlderResults(DiscoveryService source, long timestamp,
             Collection<ThingTypeUID> thingTypeUIDs) {
         HashSet<ThingUID> removedThings = new HashSet<>();
+        String sourceName = source.getClass().getName(); 
         for (DiscoveryResult discoveryResult : getAll()) {
-            if (thingTypeUIDs.contains(discoveryResult.getThingTypeUID())
-                    && discoveryResult.getTimestamp() < timestamp) {
+            String discovererName = resultDiscovererMap.get(discoveryResult); 
```

Comment:
We are within the "PersistentInbox" here. The discovererName will be null if the discoveryResult has been loaded from the storage and thus such discoveryResults will never ever be removed again. I would assume that the sourceName will need to be persisted as well in order to correctly implement this.